proto/voice-info.proto

```diff
@@ -0,0 +1,175 @@
+syntax = "proto3";
+
+package org.sim.voices;
+
+/*
+  Version info, shall adhere to semantic versioning in Hex number for easier
+  comparison.  E.g. 0x00010001 for version 1.0.1.
+  Rules are as follows:
+    - Major number shall be increased, in case of incompatible changes,
+        e.g. semantically incompatible change, new parameters needed, etc.
+    - Minor number shall be increased, in case of a compatible changes,
+        e.g. renaming of members, adding new members, changing the protobuf
+        type in a compatible way, improving a models speed/quality, etc.
+    - Patch number shall be increased, in case of a backwards-compatible
+      bug-fix
+*/
+message Revision {
+  uint32 version = 1;
+}
+
+/* Describes a voice model's file entry */
+message FileEntry {
+  enum Compression {
+    COMPRESSION_PLAIN = 0;      // no compression
+    COMPRESSION_BROTLI = 1;
+    COMPRESSION_BZIP2 = 2;
+    COMPRESSION_GZIP = 3;
+    COMPRESSION_ZIP = 4;
+  }
+  enum Format {
+    FORMAT_PLAIN = 0;           // plain file, no archive
+    FORMAT_TAR = 1;             // tar archive
+    FORMAT_CPIO = 2;            // cpio archive
+  }
+  string path = 1;                // relative path, mandatory
+  Compression compression = 2;    // file compression, optional
+  Format format = 3;              // file format, optional
+  string md5 = 4;                 // MD5 sum of file, mandatory
+}
+
+/* Speech model language / locale info */
+message Language {
+  /* <ISO 639 language code>_<ISO 3166 country code> */
+  enum Locale {
+    /* default: any language/country) */
+    LOCALE_ANY = 0;
+    is_IS = 1;
+    /* more entries can be appended here ... */
+  }
+  Locale locale = 1;  // locale of the voice, mandatory
+  string name = 2;    // language name of the voice, mandatory
+}
+
+/* Phonemes used for acoustic model, alphabets and symbol mappings */
+message PhonemeDescription {
+  enum PhonemeType {
+    PHONEME_TYPE_INVALID = 0;
+    /* use IPA symbols, these are always distinct single symbols */
+    PHONEME_TYPE_IPA = 1;
+    /* use "extended Sampas", these are mostly made up of multiple symbols */
+    PHONEME_TYPE_XSAMPA = 2;
+  }
+  /* Mapping between Alphabet and model representation */
+  message AlphabetMapping {
+    /*
+      Mapping of all IPA/XSampa symbols to their corresponding model value,
+      as well as meta-symbols like <sp> "short pause", <pau> "pause", etc. used
+      in model training.
+    */
+    map<string, uint32> symbols = 1;
+  }
```

Comment:
Should we maybe split the Alphabet Specification and special symbols used for things like `<sp>`, ... so that we can easily test the completeness of the used Alphabet on the inferencing side ? We could then also use for the special metasymbol mapping an `enum SpecialSymbols` value to a models `uint32` with appropriate documentation and by simply querying the keys of that map, we know which of these have been used at all inside the voice model itself.

Reply:
Do you mean having both a `map<string, uint3>` and `map<SpecialSymbols, uint32>` for the special symbols?

Reply:
I would definitely separate phonetic alphabets and whatever symbols are used besides phonetic symbols. For now, we are only working with pause-tags as extra symbols, that are not as rigidly defined as the phonetic alphabets - so including them in an alphabet would probably mean creating a whole new alphabet as soon as they differ in the pause-symbols, wouldn't it? Furthermore, we could be dealing with other symbols in the future

Reply:
@rkjaran : yes that is what I mean. Is there a list of the symbols already used anywhere or even a *canonical* list ? `SpecialSymbols` should be an enum, to have it well defined. @bnika: totally agree.

As the symbol representation is not so well defined as the IPA/XSAMPA list, mapping the locally used G2P special symbols to enums makes the dependencies between the differen G2P less strict. E.g. I have seen in python exist symbols starting with `@...`, whereas on Símarómur G2P we use e.g. `<sp>`.

Moreover, this SpecialSymbols map just contains the *used* symbols of the voice model, so one could tune the G2P to just produce those actually supported by the model. OTOH we have a limitation that longer pauses cannot be generated by the model itself, so probably we have to come up with a synthetic solution for this anyway.